CoEpi/repo/realm/RealmCENRepo.swift

```diff
@@ -2,54 +2,54 @@ import Foundation
 import CryptoSwift
 import RealmSwift
 
-struct CEN : Codable {
-    var CEN: String
-    var timestamp: Int64 = Int64(Date().timeIntervalSince1970)
-    var latitude: Double? = nil
-    var longitude: Double? = nil
-    
-    static func generateCENData(CENKey : String)  -> Data {
+class RealmCENRepo: CENRepo, RealmRepo {
+
+    var realm: Realm
+
+    init(realmProvider: RealmProvider) {
+        realm = realmProvider.realm
+    }
+
+    func insert(cen: CEN) -> Bool {
+        let DBCENObject = realm.objects(RealmCEN.self).filter("CEN = %@", cen.CEN)
+        if DBCENObject.count == 0 {
+            let newCEN = RealmCEN(cen)
+            try! realm.write {
+                realm.add(newCEN)
+            }
+            return true
+        } else {
+            //duplicate entry: skipping
+            return false
+        }
+    }
+
+    func loadAllCENRecords() -> [CEN]? {
+        let DBCENObject = realm.objects(RealmCEN.self).sorted(byKeyPath: "timestamp", ascending: false)
+        return DBCENObject.map { CEN(CEN: $0.CEN, timestamp: $0.timestamp) }
+    }
+
+    func generateCENData(CENKey : String) -> Data {
```

Comment:
generateCENData has nothing to do with Realm.

Reply:
Agreed, moved it to `ContactReceivedHandler`